src/runtime_src/core/common/drv/kds_core.c

```diff
@@ -255,13 +255,35 @@ kds_scu_config(struct kds_cu_mgmt *scu_mgmt, struct kds_command *xcmd)
 	return 0;
 }
 
+/**
+ * kds_test_refcnt - Determine whether the cu_refs[idx] is set
+ *
+ * return: True meaning that the cu has been set and vice verse
+ */
+static bool
+kds_test_refcnt(struct kds_client *client, int domain, u32 idx)
+{
+	u32 *refs;
+	bool is_set = false;
+
+	if (domain == DOMAIN_PL)
+		refs = client->refcnt->cu_refs;
+	else
+		refs = client->refcnt->scu_refs;
```

Comment:
The code here seems to assume that there can only be two type of domains. I'd think that it'll be better to explicitly check against domains you can handle here and return / print errors, if you see unexpected domains. Next time when we extend domain types, we'll not silently corrupt scu array.